### ISSUE TITLE
Failing checks number cache is not invalidated properly

### ISSUE THREAD

**Issue by burner1024** [WeblateOrg/weblate#1561]
### Steps to reproduce
1. Set up Redis as cache backend:
```
CACHES = {
  'default': {
    'BACKEND': 'redis_cache.RedisCache',
    'LOCATION': '127.0.0.1:6379',
    'OPTIONS': {
      'DB': 1,
    },
  },
```
2. Go into a component with some failing checks.
3. Fix some failing strings.
4. Navigate to the main component window.

### Actual behaviour
The number of strings failing checks it not updated. Running "redis-cli flushall" lets the correct number to be displayed.

### Expected behaviour
Failing checks number should be current at all times.

### Server configuration
```
 * Weblate weblate-2.15
 * Python 2.7.5
 * Django 1.10.3
 * six 1.10.0
 * social-auth-core 1.4.0
 * social-auth-app-django 1.2.0
 * django-appconf 1.0.2
 * Translate Toolkit 2.0.0
 * Whoosh 2.7.4
 * defusedxml 0.4.1
 * Git 2.13.2
 * Pillow (PIL) 1.1.7
 * dateutil 2.6.0
 * lxml 3.6.4
 * django-crispy-forms 1.6.1
 * compressor 2.1.1
 * djangorestframework 3.5.3
 * pytz 2016.7
 * pyuca N/A
 * python-bidi 0.4.0
 * pyLibravatar N/A
 * PyYAML 3.12
 * Mercurial 4.0
 * Database backends: django.db.backends.mysql
```

**Comment by nijel** [comment]
I've noticed this few times as well, but failed to reproduce at that time...

**Comment by nijel** [comment]
Fixed by c20eb330e8416ad28c41d50dcc305f4676699dce

**Comment by nijel** [comment]
Thank you for your report, the issue you have reported has just been fixed. 

* In case you see problem with the fix, please comment on this issue. 
* In case you see similar problem, please open separate issue.
* If you are happy with the outcome, consider supporting Weblate by [donating](https://weblate.org/donate/).
